client/app/src/main/java/com/microsoft/research/karya/ui/registration/OTPFragment.kt

```diff
@@ -15,17 +19,88 @@ import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.viewBinding
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpVerifyState
+import com.microsoft.research.karya.ui.registration.RegistrationViewModel.OtpSendState
+import dagger.hilt.android.AndroidEntryPoint
 
 private const val OTP_LENGTH = 6
 
+@AndroidEntryPoint
 class OTPFragment : Fragment(R.layout.fragment_o_t_p) {
 
     private val binding by viewBinding(FragmentOTPBinding::bind)
+    private val viewModel by viewModels<RegistrationViewModel>()
 
     private lateinit var registrationActivity: RegistrationActivity
     private lateinit var baseActivity: BaseActivity
     private lateinit var karyaAPI: KaryaAPIService
 
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view: View? = super.onCreateView(inflater, container, savedInstanceState)
+
+        viewModel.openDashBoardFromOTP.observe(viewLifecycleOwner, Observer { openDashBoard ->
+            if (openDashBoard) {
+                navigateToDashBoard()
+                // TODO: Make openDashboardFromOTP private and make a function call to do the above
+            }
+        })
+
+        viewModel.openProfilePictureFragmentFromOTP.observe(viewLifecycleOwner, { openProfilePictureFragment ->
+            if (openProfilePictureFragment) {
+                navigateToProfilePicture()
+                // TODO: Make openProfilePictureFragmentFromOTP private and make a function call to do the above
+            }
+        })
+
+        viewModel.currOtpVerifyState.observe(viewLifecycleOwner, { state ->
+            when (state) {
+                OtpVerifyState.SUCCESS -> onOtpVerifySuccess()
+                OtpVerifyState.FAIL -> onOtpVerifyOrResendFailure()
+                OtpVerifyState.NOT_ENTERED -> setOtpNotSentUI()
+            }
+        })
+
+        viewModel.currOtpResendState.observe(viewLifecycleOwner, { state ->
+            when (state) {
+                // TODO: Maybe indicate user after successful otp sent
+                OtpSendState.FAIL -> onOtpVerifyOrResendFailure()
+            }
+        })
```

Comment:
Can you please move all of these to a function? Just to keep the `onViewCreated` clean.

Reply:
Will do